src/textual/_compose.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .app import App
+    from .widget import Widget
+
+
+def compose(node: App | Widget) -> list[Widget]:
+    """Compose child widgets.
+
+    Args:
+        node: The parent node.
+
+    Returns:
+        A list of widgets.
+    """
+    app = node.app
+    nodes: list[Widget] = []
+    compose_stack: list[Widget] = []
+    composed: list[Widget] = []
+    app._compose_stacks.append(compose_stack)
+    app._composed.append(composed)
```

Comment:
I'm a bit lost. The lists `compose_stack` and `composed` do not seem to be modified directly in this function.
Are they being modified as a side effect of having been appended to `app._compose_stacks`/`app._composed` and the later call to `node.compose()`?

Reply:
Now, I see they are.
I don't _love_ side-effects here and they make the code harder to follow but I can't come up with a better suggestion off the top of my head.
Can you see any other possible strategies here or is this the only feasible route?

Reply:
I don't think there is any other way to make this magic work.